public/app/features/auth-config/AuthDrawer.tsx

```diff
@@ -1,53 +1,65 @@
 import { css } from '@emotion/css';
-import React, { useState } from 'react';
+import React, { JSX } from 'react';
+import { connect, ConnectedProps } from 'react-redux';
 
 import { GrafanaTheme2 } from '@grafana/data';
-import { getBackendSrv } from '@grafana/runtime';
 import { Button, Drawer, Text, TextLink, Switch, useStyles2 } from '@grafana/ui';
+import { useAppNotification } from 'app/core/copy/appNotification';
+import { StoreState } from 'app/types';
 
-export interface Props {
+import { loadSettings, saveSettings } from './state/actions';
+
+interface OwnProps {
   onClose: () => void;
 }
 
-const SETTINGS_URL = '/api/admin/settings';
-
-export const AuthDrawer = ({ onClose }: Props) => {
-  const [isOauthAllowInsecureEmailLookup, setOauthAllowInsecureEmailLookup] = useState(false);
+export type Props = OwnProps & ConnectedProps<typeof connector>;
 
-  const getSettings = async () => {
-    try {
-      const response = await getBackendSrv().get(SETTINGS_URL);
-      setOauthAllowInsecureEmailLookup(response.auth.oauth_allow_insecure_email_lookup?.toLowerCase?.() === 'true');
-    } catch (error) {}
-  };
-  const updateSettings = async (property: boolean) => {
-    try {
-      const body = {
-        updates: {
-          auth: {
-            oauth_allow_insecure_email_lookup: '' + property,
-          },
-        },
-      };
-      await getBackendSrv().put(SETTINGS_URL, body);
-    } catch (error) {}
+const mapStateToProps = (state: StoreState) => {
+  const allowInsecureEmail =
+    state.authConfig.settings?.auth?.oauth_allow_insecure_email_lookup.toLowerCase() === 'true';
+  return {
+    allowInsecureEmail,
   };
+};
 
-  const resetButtonOnClick = async () => {
-    try {
-      const body = {
-        removals: {
-          auth: ['oauth_allow_insecure_email_lookup'],
+const mapActionsToProps = {
+  loadSettings,
+  saveSettings,
+};
+
+const connector = connect(mapStateToProps, mapActionsToProps);
+
+export const AuthDrawerUnconnected = ({
+  allowInsecureEmail,
+  loadSettings,
+  onClose,
+  saveSettings,
+}: Props): JSX.Element => {
+  const notifyApp = useAppNotification();
+
+  const oauthAllowInsecureEmailLookupOnChange = async () => {
+    saveSettings({
+      updates: {
+        auth: {
+          oauth_allow_insecure_email_lookup: '' + !allowInsecureEmail,
         },
-      };
-      await getBackendSrv().put(SETTINGS_URL, body);
-      getSettings();
-    } catch (error) {}
+      },
+    })
+      .then(() => loadSettings(false))
```

Comment:
I would recommend using async/await approach, it produces much clearer code.
```ts
try {
  await saveSettings()
  await loadSettings(false)
  notifyApp.success('Settings saved')
} catch (error) {
  notifyApp.error('Failed to save settings')
}
```

You also could put `loadSettings()` call into `saveSettings()` action if you always call it after (I think you in most of cases need to re-fetch settings after saving it to display updated values).

Reply:
I prefer promises over async calls, but this is just me. I have changed the code as per your suggestions.

Reply:
I decided not to include `loadSettings` inside `saveSettings` because I'm aware they're being used by the old LDAP and SAML UI. I rather keep this clean for now.

Do you think is necessary?